metrics/web/docs/features/builds_aggregation/01_builds_aggregation_design.md

```diff
@@ -0,0 +1,53 @@
+
+## References
+
+> Link to supporting documentation, GitHub tickets, etc.
+
+- [Project Metrics Definitions](https://github.com/platform-platform/monorepo/blob/master/docs/05_project_metrics.md)
+- [Github epic: Reduce Firebase usage / document reads](https://github.com/platform-platform/monorepo/issues/1042)
+- [Firebase builds aggregation](https://github.com/platform-platform/monorepo/blob/master/metrics/firebase/docs/features/builds_aggregation/01_firestore_builds_aggregation_design.md)
+
+## Metrics application
+
+The following sub-sections provide an implementation of Build aggregation integration for the Metrics Web Application by layers. Read more about layers and their responsibilities in the [Metrics Web Application architecture document](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/01_metrics_web_application_architecture.md).
+
+### Domain layer
+
+The domain layer should provide an interface for the `FirestoreBuildDayRepository`. Also, the layer provides a usecase required to interact with the repository, and entity for the `Builds Aggregation` feature. Thus, the following list of classes should be implemented to fit the feature requirements:
+
+- Implement the `BuildDayRepository` interface with appropriate methods.
+- Add the `BuildDay` entity with fields that come from a remote API.
+- Add the `ReceiveBuildDayProjectMetricsUpdates` usecase with the `BuildDayParam`.
+
+The following class diagram demonstrates the domain layer structure:
+
+![Domain layer diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/metrics_app_builds_aggregation_doc/metrics/web/docs/features/builds_aggregation/diagrams/build_days_domain_layer_class_diagram.puml)
+
+### Data layer
+
+The data layer provides the `FirestoreBuildDayRepository` implementation of `BuildDayRepository` and `BuildDayData` that represents a `DataModel` implementation for the `BuildDay` entity.
+
+![Data layer diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/metrics_app_builds_aggregation_doc/metrics/web/docs/features/builds_aggregation/diagrams/build_days_data_layer_class_diagram.puml)
+
+### Presentation layer
+
+Once we've added both the domain and data layers, it's time to add the feature to the presentation.
+
+The `ProjectMetricsNotifier` should have one more argument - `ReceiveBuildDayProjectMetricsUpdates` usecase. It provides a `BuildDayProjectMetrics` to display on the dashboard.
```

Comment:
Would it be great to describe in a bit more detail what changes are required in the `ProjectMetricsNotifier` to implement this feature? I think it would be useful to mention all new/updated methods on the class diagram.